internal/bminventory/inventory.go

```diff
@@ -437,6 +437,17 @@ func (b *bareMetalInventory) UpdateDiscoveryIgnition(ctx context.Context, params
 		return installer.NewUpdateDiscoveryIgnitionInternalServerError().WithPayload(common.GenerateError(http.StatusInternalServerError, err))
 	}
 
+	exists, err := b.objectHandler.DoesObjectExist(ctx, getImageName(*c.ID))
+	if err != nil {
+		return common.NewApiError(http.StatusInternalServerError, err)
+	}
+	if exists {
+		if err := b.objectHandler.DeleteObject(ctx, getImageName(*c.ID)); err != nil {
+			return common.NewApiError(http.StatusInternalServerError, err)
```

Comment:
Do we actually want to fail this call in this case (and also the one above)?
The edit was successfully validated and saved...

@avishayt ?

Reply:
I guess we do, otherwise we will have weird behavior.  Maybe we should put this code before the DB update?